receiver/otlpreceiver/config.go

```diff
@@ -82,3 +83,7 @@ func (cfg *Config) Unmarshal(componentParser *confmap.Conf) error {
 
 	return nil
 }
+
+func (cfg *Config) Marshal(enc configencoder.Encoder) (interface{}, error) {
+	return enc.Encode(cfg)
+}
```

Comment:
This is required because I added the `Marshal` function to the `Unmarshallable` interface to force configs that implement `Unmarshal` to implement the `Marshal` function. In the case of the otlp receiver, it can be encoded as is since the `Unmarshal` function simply nils out unset protocols, which can be retained when marshalling with `omitempty`.

Reply:
Separated the `Marshal` function into a `Marshallable` interface.